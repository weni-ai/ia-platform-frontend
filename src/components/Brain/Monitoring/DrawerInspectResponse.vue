<template>
  <UnnnicDrawer
    :modelValue="modelValue"
    :title="$t('router.monitoring.inspect_response.title')"
    :description="
      $t(
        'router.monitoring.inspect_response.intelligent_agent_response_details',
      )
    "
    @close="close"
  >
    <template #content>
      <section class="inspect-response">
        <section class="inspect-response__contact">
          <UnnnicIntelligenceText
            color="neutral-cloudy"
            family="secondary"
            size="body-gt"
            tag="p"
          >
            {{ $t('router.monitoring.inspect_response.contact_message') }}:
          </UnnnicIntelligenceText>
          <p class="contact__text">“{{ inspectionData.text }}”</p>
        </section>
        <section class="inspect-response__agent">
          <UnnnicIntelligenceText
            color="neutral-cloudy"
            family="secondary"
            size="body-gt"
            tag="p"
          >
            {{ $t('router.monitoring.inspect_response.agent_response') }}:
          </UnnnicIntelligenceText>
          <p
            v-if="inspectionData.llm.status !== 'success'"
            class="agent__text"
          >
            “{{ inspectionData.llm.response }}”
          </p>
          <section v-else>
            <template
              v-for="fragment of groundednessFragments"
              :key="fragment.sentence"
            >
              <p
                :class="[
                  'agent__text',
                  'agent__text--underlined',
                  `agent__text--underlined-${getReliabilityLevel(fragment)}`,
                ]"
              >
                {{ fragment.sentence }}
              </p>
            </template>
          </section>
        </section>
      </section>
    </template>
  </UnnnicDrawer>
</template>

<script setup>
import { computed } from 'vue';

const props = defineProps({
  modelValue: {
    type: Boolean,
    default: false,
  },

  inspectionData: {
    type: Object,
    required: true,
  },
});

const emit = defineEmits(['update:model-value']);

const groundednessFragments = computed(() => {
  const data = props.inspectionData.groundedness.map((fragment) => ({
    ...fragment, // Required to create copies of the fragments and not update the prop
  }));

  if (data.length > 0) {
    const firstFragment = data[0];
    const lastFragment = data.at(-1);

    firstFragment.sentence = `“${firstFragment.sentence}`;
    lastFragment.sentence = `${lastFragment.sentence}”`;
  }

  return data;
});

function close() {
  emit('update:model-value', false);
}

function getReliabilityLevel(item) {
  const { score } = item;

  if (score === 0) return 'unreliable';
  if (score <= 33) return 'slightly-reliable';
  if (score <= 66) return 'moderately-reliable';
  if (score <= 100) return 'highly-reliable';
  return 'unreliable';
}
</script>

<style lang="scss" scoped>
.inspect-response {
  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-md;

  &__contact,
  &__agent {
    display: grid;
    gap: $unnnic-spacing-nano;

    .contact__text,
    .agent__text {
      color: $unnnic-color-neutral-darkest;
      font-size: $unnnic-font-size-body-lg;
    }
  }

  &__agent {
    .agent__text {
      display: inline;

      line-height: $unnnic-font-size-body-lg + $unnnic-line-height-md;

      &--underlined {
        margin-right: $unnnic-spacing-nano;

        text-decoration: underline;
        text-decoration-color: $unnnic-color-neutral-cleanest;
        text-decoration-thickness: 3px;
        text-underline-offset: 4px;

        &-slightly-reliable {
          text-decoration-color: $unnnic-color-weni-300;
        }
        &-moderately-reliable {
          text-decoration-color: $unnnic-color-weni-600;
        }
        &-highly-reliable {
          text-decoration-color: $unnnic-color-weni-800;
        }
      }
    }
  }
}
</style>
