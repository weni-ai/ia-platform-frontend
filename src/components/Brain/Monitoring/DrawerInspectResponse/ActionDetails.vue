<template>
  <section
    data-testid="action-details-container"
    class="action-details"
  >
    <UnnnicAvatarIcon
      data-testid="action-icon"
      class="action__icon"
      size="sm"
      icon="bolt"
      scheme="aux-blue"
    />
    <p
      data-testid="action-name"
      class="action__name"
    >
      {{ action.name }}
    </p>
  </section>
</template>

<script setup>
defineProps({
  action: {
    type: Object,
    required: true,
  },
});
</script>

<style scoped lang="scss">
.action-details {
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;
  border-radius: $unnnic-border-radius-sm;

  padding: $unnnic-spacing-xs;

  display: flex;
  align-items: center;
  gap: $unnnic-spacing-xs;

  .action__name {
    color: $unnnic-color-neutral-dark;
    font-size: $unnnic-font-size-body-gt;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }
}
</style>
