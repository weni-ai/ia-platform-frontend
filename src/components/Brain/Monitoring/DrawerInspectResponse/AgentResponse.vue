<template>
  <section class="inspect-response__agent">
    <UnnnicIntelligenceText
      data-testid="agent-response-title"
      color="neutral-cloudy"
      family="secondary"
      size="body-gt"
      tag="p"
    >
      {{ $t('router.monitoring.inspect_response.agent_response') }}:
    </UnnnicIntelligenceText>
    <ResponseSuccessGroundedness
      v-if="showSuccessGroundedness"
      data-testid="agent-response-success-groundedness"
      :inspectionData="inspectionData"
    />
    <p
      v-else
      data-testid="agent-response-text"
      class="agent__text"
    >
      “{{ response }}”
    </p>
  </section>
</template>

<script setup>
import { computed } from 'vue';

import ResponseSuccessGroundedness from '../ResponseSuccessGroundedness/index.vue';

const props = defineProps({
  status: {
    type: String,
    required: true,
  },
  response: {
    type: String,
    default: '',
  },
  inspectionData: {
    type: Object,
    required: true,
  },
});

const showSuccessGroundedness = computed(
  () =>
    props.status === 'success' &&
    props.inspectionData.groundedness.find(
      (fragment) => fragment.sources.length,
    ),
);
</script>
