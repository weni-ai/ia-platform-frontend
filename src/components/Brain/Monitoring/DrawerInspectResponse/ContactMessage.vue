<template>
  <section class="inspect-response__contact">
    <UnnnicIntelligenceText
      color="neutral-cloudy"
      family="secondary"
      size="body-gt"
      tag="p"
    >
      {{ $t('router.monitoring.inspect_response.contact_message') }}:
    </UnnnicIntelligenceText>
    <p class="contact__text">“{{ text }}”</p>
  </section>
</template>

<script setup>
defineProps({
  text: {
    type: String,
    required: true,
  },
});
</script>
