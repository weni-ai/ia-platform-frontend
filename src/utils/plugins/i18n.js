import * as VueI18n from 'vue-i18n';

import en from '../../locales/en.json';
import ptbr from '../../locales/pt_br.json';
import es from '../../locales/es.json';

const languages = {
  'en-US': en,
  'pt-BR': ptbr,
  es,
};

const numberFormats = {
  'en-US': {
    currency: {
      style: 'currency',
      currency: 'BRL',
      currencyDisplay: 'symbol',
    },
  },
  'pt-BR': {
    currency: {
      style: 'currency',
      currency: 'BRL',
      currencyDisplay: 'symbol',
    },
  },
  es: {
    currency: {
      style: 'currency',
      currency: 'BRL',
      currencyDisplay: 'symbol',
    },
  },
};

const messages = Object.assign(languages);

const i18n = VueI18n.createI18n({
  locale: navigator.language || navigator.userLanguage,
  fallbackLocale: 'en-US',
  messages,
  numberFormats,
});

export default i18n;
