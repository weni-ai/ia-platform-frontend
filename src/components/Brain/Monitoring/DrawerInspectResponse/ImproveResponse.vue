<template>
  <section class="improve-response">
    <header
      v-if="type === 'failed'"
      class="improve-response__header-failed"
      data-testid="header-failed"
    >
      <UnnnicIcon
        icon="cancel"
        scheme="aux-red-500"
        size="sm"
      />
      <UnnnicIntelligenceText
        color="aux-red-500"
        family="secondary"
        size="body-gt"
        tag="p"
      >
        {{ $t('router.monitoring.improve_response.agent_unable_respond') }}
      </UnnnicIntelligenceText>
    </header>
    <section
      :class="[
        'improve-response__actions',
        { 'improve-response__actions--with-header': type === 'failed' },
      ]"
    >
      <UnnnicIntelligenceText
        color="neutral-cloudy"
        family="secondary"
        size="body-gt"
        tag="p"
      >
        {{ $t('router.monitoring.improve_response.title') }}
      </UnnnicIntelligenceText>

      <section class="actions__buttons">
        <UnnnicButton
          v-if="actionToEdit"
          data-testid="edit-action-button"
          size="small"
          :text="
            $t('router.monitoring.improve_response.edit_action', {
              name: actionToEdit.name,
            })
          "
          type="secondary"
          @click="openEditModalAction"
        />
        <UnnnicButton
          v-else
          data-testid="add-content-button"
          size="small"
          :text="$t('router.monitoring.improve_response.add_new_content')"
          type="secondary"
          @click="isModalAddContentOpen = true"
        />
        <UnnnicButton
          data-testid="add-action-button"
          size="small"
          :text="$t('router.monitoring.improve_response.add_new_action')"
          type="secondary"
          @click="isModalAddActionOpen = true"
        />
      </section>
    </section>

    <ModalAddContent
      v-if="isModalAddContentOpen"
      v-model="isModalAddContentOpen"
      data-testid="modal-add-content"
    />

    <ModalActions
      v-if="isModalAddActionOpen"
      v-model="isModalAddActionOpen"
      data-testid="modal-add-action"
      actionGroup="custom"
      :actionToEditUuid="shouldEditAction ? actionToEdit?.uuid : undefined"
      @previous-step="isModalAddActionOpen = false"
    />
  </section>
</template>

<script setup>
import ModalAddContent from '../ModalAddContent/index.vue';

import { computed, onMounted, ref, watch } from 'vue';
import { useMonitoringStore } from '@/store/Monitoring';
import { useActionsStore } from '@/store/Actions';

import ModalActions from '@/components/actions/ModalActions.vue';

const props = defineProps({
  type: {
    type: String,
    required: true,
    validator: (value) => ['action', 'failed', 'success'].includes(value),
  },
});

const monitoringStore = useMonitoringStore();
const actionsStore = useActionsStore();

const actionToEdit = computed(
  () => monitoringStore.messages.inspectedAnswer?.action,
);

const isModalAddContentOpen = ref(false);
const isModalAddActionOpen = ref(false);
const shouldEditAction = ref(false);

function openEditModalAction() {
  isModalAddActionOpen.value = true;
  shouldEditAction.value = true;
}

watch(isModalAddActionOpen, (newIsModalAddActionOpen) => {
  if (!newIsModalAddActionOpen) {
    shouldEditAction.value = false;
  }
});

onMounted(() => {
  if (actionsStore.actions.status !== 'complete') {
    actionsStore.load();
  }
});
</script>

<style scoped lang="scss">
.improve-response {
  &__header-failed {
    border-radius: $unnnic-border-radius-sm $unnnic-border-radius-sm 0 0;

    padding: $unnnic-spacing-sm;

    background-color: $unnnic-color-aux-red-100;

    display: flex;
    gap: $unnnic-spacing-xs;
    align-items: center;
  }

  &__actions {
    border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;
    border-radius: $unnnic-border-radius-sm;
    padding: $unnnic-spacing-sm;

    display: grid;
    gap: $unnnic-spacing-xs;

    &--with-header {
      border-radius: 0 0 $unnnic-border-radius-sm $unnnic-border-radius-sm;
      border-top: none;
    }

    .actions__buttons {
      display: grid;
      gap: $unnnic-spacing-nano;
    }
  }
}
</style>
