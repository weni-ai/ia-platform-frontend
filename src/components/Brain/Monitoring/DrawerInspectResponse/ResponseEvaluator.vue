<template>
  <section class="response-evaluator">
    <UnnnicDivider
      data-testid="divider"
      class="response-evaluator__divider"
      ySpacing="md"
    />

    <UnnnicIntelligenceText
      data-testid="rate-the-answer"
      class="response-evaluator__rate-the-answer"
      color="neutral-dark"
      family="secondary"
      size="body-gt"
      tag="p"
    >
      {{ $t('router.monitoring.inspect_response.rate_the_answer') }}:
    </UnnnicIntelligenceText>

    <section class="response-evaluator__actions">
      <button
        data-testid="like-button"
        :class="['actions__button', { 'actions__button--bold': isApproved }]"
        @click="setIsApproved(true)"
      >
        <UnnnicIcon
          icon="thumb_up"
          scheme="neutral-cloudy"
          size="sm"
          :filled="isApproved"
        />{{ $t('router.monitoring.inspect_response.i_liked') }}
      </button>
      <button
        data-testid="dislike-button"
        :class="[
          'actions__button',
          { 'actions__button--bold': isApproved === false },
        ]"
        @click="setIsApproved(false)"
      >
        <UnnnicIcon
          icon="thumb_down"
          scheme="neutral-cloudy"
          size="sm"
          :filled="isApproved === false"
        />{{ $t('router.monitoring.inspect_response.i_didnt_like') }}
      </button>
    </section>

    <ImproveResponse
      v-if="isApproved === false"
      class="response-evaluator__improve-response"
      type="action"
    />
  </section>
</template>

<script setup>
import { ref } from 'vue';
import { useMonitoringStore } from '@/store/Monitoring';

import ImproveResponse from './ImproveResponse.vue';

const props = defineProps({
  isApproved: {
    type: [Boolean, null],
    default: null,
  },
});

const monitoringStore = useMonitoringStore();

const isApproved = ref(props.isApproved);

async function setIsApproved(boolean) {
  const oldIsApproved = isApproved.value;
  isApproved.value = boolean;

  try {
    await monitoringStore.rateAnswer({ is_approved: boolean });
  } catch (error) {
    console.log('error', error);

    isApproved.value = oldIsApproved;
  }
}
</script>

<style lang="scss" scoped>
.response-evaluator {
  display: flex;
  flex-direction: column;

  .response-evaluator__divider {
    margin-top: 0;
  }

  &__rate-the-answer {
    margin-bottom: $unnnic-spacing-xs;
  }

  &__actions {
    display: flex;
    gap: $unnnic-spacing-sm;

    .actions__button {
      padding: 0;
      border: none;
      background: none;

      cursor: pointer;

      display: flex;
      align-items: center;
      justify-content: center;
      gap: $unnnic-spacing-nano;

      color: $unnnic-color-neutral-cloudy;
      font-family: $unnnic-font-family-secondary;
      font-size: $unnnic-font-size-body-md;
      line-height: $unnnic-font-size-body-md + $unnnic-line-height-md;

      &--bold {
        font-weight: $unnnic-font-weight-bold;
      }
    }
  }

  &__improve-response {
    margin-top: $unnnic-spacing-sm;
  }
}
</style>
