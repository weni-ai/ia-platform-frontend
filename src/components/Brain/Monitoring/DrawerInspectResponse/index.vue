<template>
  <UnnnicDrawer
    data-testid="drawer"
    :modelValue="modelValue"
    :title="$t('router.monitoring.inspect_response.title')"
    :description="
      $t(
        'router.monitoring.inspect_response.intelligent_agent_response_details',
      )
    "
    @close="close"
  >
    <template #content>
      <section class="inspect-response">
        <ContactMessage :text="inspectionData.text" />
        <ActionDetails
          v-if="llmStatus === 'action'"
          :action="inspectionData.action"
        />
        <AgentResponse
          v-if="llmStatus !== 'action'"
          :status="llmStatus"
          :response="inspectionData.llm.response"
          :inspectionData="inspectionData"
        />
        <ResponseEvaluator
          v-if="llmStatus !== 'failed'"
          :isApproved="inspectionData.is_approved"
        />
      </section>
    </template>
  </UnnnicDrawer>
</template>

<script setup>
import { computed } from 'vue';

import ContactMessage from './ContactMessage.vue';
import ActionDetails from './ActionDetails.vue';
import AgentResponse from './AgentResponse.vue';
import ResponseEvaluator from './ResponseEvaluator.vue';

const props = defineProps({
  modelValue: {
    type: Boolean,
    default: false,
  },
  inspectionData: {
    type: Object,
    required: true,
  },
});

const emit = defineEmits(['update:model-value']);

const llmStatus = computed(() => props.inspectionData.llm.status);

function close() {
  emit('update:model-value', false);
}
</script>

<style lang="scss" scoped>
.inspect-response {
  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-md;

  &__contact,
  &__agent {
    display: grid;
    gap: $unnnic-spacing-nano;

    .contact__text,
    .agent__text {
      color: $unnnic-color-neutral-darkest;
      font-size: $unnnic-font-size-body-lg;
    }
  }

  &__agent .agent__text {
    display: inline;
    line-height: $unnnic-font-size-body-lg + $unnnic-line-height-md;
  }
}
</style>
