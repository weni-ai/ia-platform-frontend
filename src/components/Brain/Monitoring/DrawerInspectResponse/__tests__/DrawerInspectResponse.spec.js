import { describe, it, expect, beforeEach } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import i18n from '@/utils/plugins/i18n';

import DrawerInspectResponse from '../index.vue';
import ContactMessage from '../ContactMessage.vue';
import Unnnic from '@weni/unnnic-system';
import ActionDetails from '../ActionDetails.vue';
import AgentResponse from '../AgentResponse.vue';
import ResponseEvaluator from '../ResponseEvaluator.vue';

describe('DrawerInspectResponse', () => {
  let wrapper;
  const inspectionDataMock = {
    text: 'Test message',
    llm: { status: 'action', response: 'Sample response' },
    action: { name: 'action-name' },
    is_approved: true,
  };

  const contextT = (key) =>
    i18n.global.t(`router.monitoring.inspect_response.${key}`);

  const mountWrapper = (props = {}) => {
    wrapper = shallowMount(DrawerInspectResponse, {
      props: {
        modelValue: true,
        inspectionData: inspectionDataMock,
        ...props,
      },
      global: {
        stubs: {
          UnnnicDrawer: Unnnic.unnnicDrawer,
        },
      },
    });
  };

  beforeEach(() => {
    mountWrapper();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders the drawer with the correct props', () => {
    const drawer = wrapper.findComponent('[data-testid="drawer"]');
    expect(drawer.exists()).toBe(true);
    expect(drawer.props()).toMatchObject({
      modelValue: true,
      title: contextT('title'),
      description: contextT('intelligent_agent_response_details'),
    });
  });

  it('renders the ContactMessage component with the correct props', () => {
    const contactMessage = wrapper.findComponent(ContactMessage);
    expect(contactMessage.exists()).toBe(true);
    expect(contactMessage.props('text')).toEqual(inspectionDataMock.text);
  });

  it('conditionally renders the ActionDetails component when llm.status is "action"', () => {
    const actionDetails = wrapper.findComponent(ActionDetails);
    expect(actionDetails.exists()).toBe(true);

    expect(actionDetails.props('action')).toEqual(inspectionDataMock.action);
  });

  it('does not render the ActionDetails component when llm.status is not "action"', async () => {
    await wrapper.setProps({
      inspectionData: {
        ...inspectionDataMock,
        llm: { ...inspectionDataMock.llm, status: 'success' },
      },
    });
    const actionDetails = wrapper.findComponent(ActionDetails);
    expect(actionDetails.exists()).toBe(false);
  });

  it('renders the AgentResponse component with the correct props', () => {
    const agentResponse = wrapper.findComponent(AgentResponse);
    expect(agentResponse.exists()).toBe(true);
    expect(agentResponse.props()).toMatchObject({
      status: inspectionDataMock.llm.status,
      response: inspectionDataMock.llm.response,
      inspectionData: inspectionDataMock,
    });
  });

  it('does not render the ResponseEvaluator component when llm.status is "failed"', async () => {
    await wrapper.setProps({
      inspectionData: {
        ...inspectionDataMock,
        llm: { ...inspectionDataMock.llm, status: 'failed' },
      },
    });
    const responseEvaluator = wrapper.findComponent(ResponseEvaluator);
    expect(responseEvaluator.exists()).toBe(false);
  });

  it('conditionally renders the ResponseEvaluator component when llm.status is not "failed"', () => {
    const responseEvaluator = wrapper.findComponent(ResponseEvaluator);
    expect(responseEvaluator.exists()).toBe(true);
    expect(responseEvaluator.props('isApproved')).toEqual(
      inspectionDataMock.is_approved,
    );
  });

  it('emits "update:model-value" with false when close is called', async () => {
    const drawer = wrapper.findComponent('[data-testid="drawer"]');
    await drawer.vm.$emit('close');
    expect(wrapper.emitted('update:model-value')).toBeTruthy();
    expect(wrapper.emitted('update:model-value')[0]).toEqual([false]);
  });
});
