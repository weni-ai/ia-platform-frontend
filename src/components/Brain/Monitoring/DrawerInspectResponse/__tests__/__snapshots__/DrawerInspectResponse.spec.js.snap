// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DrawerInspectResponse > matches the snapshot 1`] = `
"<aside data-v-9f445947="" data-v-9f3b4af2="" class="unnnic-drawer" data-testid="drawer">
  <section data-v-9f445947="" class="unnnic-drawer__overlay" data-testid="overlay"></section>
  <transition-stub data-v-9f445947="" name="drawer" appear="true" persisted="false" css="true">
    <section data-v-9f445947="" data-testid="drawer-container" class="unnnic-drawer__container unnnic-drawer__container--md">
      <header data-v-9f445947="" class="unnnic-drawer__header">
        <section data-v-9f445947="" class="unnnic-drawer__title-container">
          <h1 data-v-9f445947="" class="unnnic-drawer__title" data-testid="drawer-title">Inspect response</h1>
          <p data-v-9f445947="" class="unnnic-drawer__description" data-testid="drawer-description">Intelligent agent response details</p>
        </section>
        <unnnic-icon-stub data-v-9f445947="" filled="false" next="false" icon="arrow_back" clickable="true" size="avatar-nano" scheme="neutral-darkest" class="unnnic-drawer__close" data-testid="close-icon"></unnnic-icon-stub>
      </header>
      <section data-v-9f445947="" class="unnnic-drawer__content">
        <section data-v-9f3b4af2="" class="inspect-response">
          <contact-message-stub data-v-9f3b4af2="" text="Test message"></contact-message-stub>
          <action-details-stub data-v-9f3b4af2="" action="[object Object]"></action-details-stub>
          <agent-response-stub data-v-9f3b4af2="" status="action" inspectiondata="[object Object]" response="Sample response"></agent-response-stub>
          <response-evaluator-stub data-v-9f3b4af2="" isapproved="true"></response-evaluator-stub>
        </section>
      </section>
      <!---->
    </section>
  </transition-stub>
</aside>"
`;
